src/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareParameters.cs

```diff
@@ -182,6 +182,18 @@ public DateTimeOffset GetEndOrDefault(DateTimeOffset defaultValue)
         return parameters;
     }
 
+    /// <summary>
+    /// Get an IEnumerable of all the PropertyNames defined in the CarbonAwareParameters class
+    /// </summary>
+    public static IEnumerable<PropertyName> GetPropertyNames() {
+        var properties = new List<PropertyName>();
+        foreach (string name in Enum.GetNames<PropertyName>())
```

Comment:
A one-liner might accomplish the conversion.

```suggestion
        return Enum.GetNames(typeof(PropertyName)).ToList();
```